src/read.c

```diff
@@ -626,13 +626,11 @@ static LHSRef ReadVar(ScannerState * s, TypSymbolSet follow)
     // try to look up the variable on the stack of local variables
     const UInt countNams = LEN_PLIST(ReaderState()->StackNams);
     for (nest = 0; nest < countNams; nest++) {
-#ifndef SYS_IS_64_BIT
```

Comment:
Out of curiosity, what is the reason for changing this now or why was this only enforced in 32bit systems in the first place?

Reply:
The limit used to be enforced uniformly but with a incorrect error message up to and including GAP 4.6:
```C
...
	if (nest >= 65536)
	  {
	    Pr("Warning: abandoning search for %s at 1024th higher frame\n",
	       (Int)Value,0L);
	    break;
	  }
      }
      lvars0 = PTR_BAG( lvars0 )[2];
      nest0++;
	if (nest0 >= 65536)
	  {
	    Pr("Warning: abandoning search for %s 4096 frames up stack\n",
	       (Int)Value,0L);
	    break;
	  }
...
```
In GAP 4.7 we fixed the messages to say it 65536 in both cases.

in GAP 4.9 the code underwent a major refactoring. The code became clearer, and it became obvious were the restriction to 16 bits came from: to make sure the values `nest` and `indx` fit together into a 32bit value. But on 32bit systems, `nest` actually could take 48=64-16 bits, which is for all practical purposes unlimited. Hence the check was limited to happen on 32bit systems.

However, later on (basically in GAP 4.11), I changed struct `LHSRef` and limited the storage for this combined value back to 32bit (for various reasons, IMHO good ones, but doesn't matter for now). Anyway, this means we should limit the value on 64 bit systems, too!

There is also a non-technical argument: it seems bad to allow code to work on 64bit that won't work on 32bit. Not that it is likely: code that breaks this limit would be extremely weird (and slow).

Reply:
I'll update the commit message for this change, and also for the other thing you pointed out.

Reply:
Since you’ve put this comment into a commit message, it may as well be correct, so I offer the following suggestions:

- ‘A incorrect’ → ‘an incorrect’
- ‘Obvious were’ → ‘obvious where’
- ‘But on 32bit systems’ → ‘but on 64bit systems’

I also think that 32- and 64-bit should be written with hyphens, but that’s a totally inconsequential nitpick, and so I’m not asking you to change your usage!

Reply:
@wilfwilson thanks, all fixed and some more :-)

Reply:
Thanks for explaining @fingolfin !